xscen/aggregate.py

```diff
@@ -334,36 +335,70 @@ def spatial_mean(
     xarray.Dataset.mean, xarray.Dataset.interp, xesmf.SpatialAverager
     """
     kwargs = kwargs or {}
+    if method == "mean":
+        warnings.warn(
+            "xs.spatial_mean with method=='mean' is deprecated and will be abandoned in a future release. Use method=='cos-lat' instead for a more robust but similar method.",
+            category=DeprecationWarning,
+            stacklevel=2,
+        )
+    elif method == "interp_coord":
+        warnings.warn(
+            "xs.spatial_mean with method=='interp_coord' is deprecated. Use method=='interp_centroid' instead.",
+            category=DeprecationWarning,
+            stacklevel=2,
+        )
+        method = "interp_centroid"
 
     # If requested, call xscen.extraction.clisops_subset prior to averaging
     if call_clisops:
         ds = clisops_subset(ds, region)
 
+    if method == "cos-lat":
+        if "latitude" not in ds.cf.coordinates:
+            raise ValueError(
+                "Could not determine the latitude name using CF conventions. "
+                "Use kwargs = {lat: str} to specify the coordinate name."
+            )
+
+        if "units" not in ds.cf["latitude"].attrs:
+            logger.warning(
+                f"{ds.attrs.get('cat:id', '')}: Latitude does not appear to have units. Make sure that the computation is right."
+            )
+        elif ds.cf["latitude"].attrs["units"] != "degrees_north":
+            logger.warning(
+                f"{ds.attrs.get('cat:id', '')}: Latitude units is '{ds.cf['latitude'].attrs['units']}', expected 'degrees_north'. "
+                f"Make sure that the computation is right."
+            )
+
+        if (ds.cf["longitude"].min() < -160) & (ds.cf["longitude"].max() > 160):
+            logger.warning(
+                "The region appears to be crossing the -180/180° meridian. Bounds computation is currently bugged in cf_xarray. "
+                "Make sure that the computation is right."
+            )
+
+        ds = ds.cf.add_bounds(["longitude", "latitude"])
+        weights = xr.DataArray(
+            pygeos.area(
+                pygeos.polygons(pygeos.linearrings(ds.lon_bounds, ds.lat_bounds))
+            ),
+            dims=ds.cf["longitude"].dims,
+            coords=ds.cf["longitude"].coords,
+        ) * np.cos(np.deg2rad(ds.cf["latitude"]))
+
+        ds_agg = ds.weighted(weights).mean(
+            [d for d in ds.cf.axes["X"] + ds.cf.axes["Y"]], keep_attrs=True
+        )
+
+        # Prepare the History field
+        new_history = (
+            f"[{datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] "
+            f"weighted mean(dim={[d for d in ds.cf.axes['X'] + ds.cf.axes['Y']]}) using a 'cos-lat' approximation"
```

Comment:
I'm not sure how to phrase it, but I feel this should mention that we are also weighting by the cell area in deg².

Reply:
`using a 'cos-lat' approximation of areacella (in deg2)` ?

Reply:
Agreed!